### PR TITLE
feature: improve error handling on creating SmartDocuments DOC file

### PR DESCRIPTION
**Pull request by hsiliev**
Return 400 error instead of 500 when no template mapping exists

Solves: PZ-3643

### CODE REVIEW

File: src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt [hunk 1]
```diff
@@ -154,31 +156,40 @@ class SmartDocumentsTemplatesService @Inject constructor(
...
-                                        root.get<SmartDocumentsTemplate>(SmartDocumentsTemplate::smartDocumentsId.name),
-                                        templateId
                                     )
                                 )
-                            )
-                    ).singleResult.informatieObjectTypeUUID
+                        ).singleResult.informatieObjectTypeUUID
+                    } catch (nre: NoResultException) {
+                        throw SmartDocumentsException(
+                            "No information object type mapped for " +
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt:189]: We should always include the original exception in this new exception I think. At least in Kotlin you will get a Detekt error if you don't which is right.

File: src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt [hunk 2]
```diff
@@ -154,31 +156,40 @@ class SmartDocumentsTemplatesService @Inject constructor(
...
-                                    ),
-                                    builder.equal(
-                                        root.get<SmartDocumentsTemplate>(SmartDocumentsTemplate::smartDocumentsId.name),
-                                        templateId
                                     )
                                 )
-                            )
-                    ).singleResult.informatieObjectTypeUUID
+                        ).singleResult.informatieObjectTypeUUID
+                    } catch (nre: NoResultException) {
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt:187]:
```suggestion
                    } catch (noResultsException: NoResultException) {
```

I don't like abbreviations when there is no clear benefit. ;-)

File: src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt [hunk 3]
```diff
@@ -190,21 +201,28 @@ class SmartDocumentsTemplatesService @Inject constructor(
...
+                                        root.get<String>(SmartDocumentsTemplateGroup::smartDocumentsId.name),
+                                        templateGroupId
+                                    )
                                 )
-                            )
-                    ).singleResult.name
+                        ).singleResult.name
+                    } catch (nre: NoResultException) {
+                        throw SmartDocumentsException(
+                            "No information object type mapped for template group id $templateGroupId"
```
Comment (edgarvonk) [inline src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt:223]: Same here. We should include the original exception here. Or log it.

**Review by edgarvonk — CHANGES_REQUESTED** [review]
looks good. some small comments

File: src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt (hunk 1 above)
Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt]: Logged with FINE

Comment (edgarvonk) [reply src/main/kotlin/net/atos/zac/smartdocuments/SmartDocumentsTemplatesService.kt]: I think that's ok in this case. :-)

**Review by edgarvonk — APPROVED** [review]

**Pull request merged**